src/data_generators/homogeneous_impression_generator.py

```diff
@@ -0,0 +1,67 @@
+# Copyright 2021 The Private Cardinality Estimation Framework Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Generate a stream of impressions from a poisson distr with fixed lambda."""
+
+from typing import List
+from numpy.random import RandomState
+
+from wfa_planning_evaluation_framework.data_generators.impression_generator import (
+    ImpressionGenerator,
+)
+
+
+class HomogeneousImpressionGenerator(ImpressionGenerator):
+    """Generate a random sequence of viewer id's of ad impressions.
+
+    This class, along with PricingGenerator, assists in the generation of
+    random PublisherDataFiles.  The ImpressionGenerator will generate a
+    sequence of random impressions according to specified criteria.
+    """
+
+    def __init__(self, n: int, poisson_lambda: float, random_state: RandomState = None):
+        """Constructor for the HomogeneousImpressionGenerator.
+
+        For each user, the number of impressions assigned to that user is
```

Comment:
Mention that you are shifting Poisson by 1?

Reply:
Thanks.  Done.